### PR TITLE
Enable maps of non-nilable owned and borrowed

### PR DESCRIPTION
**Pull request by daviditen**
Remove the error for non-nilable map of owned. It is now working at least
well enough to pass the "types" tests.

Add lifetime annotations for a few map methods to get them working with
borrowed classes.

Remove .futures and .bads for tests that are now passing.

### CODE REVIEW

File: modules/standard/Map.chpl [hunk 1]
```diff
@@ -469,7 +461,7 @@ module Map {
                `false` otherwise.
      :rtype: bool
     */
-    proc add(in k: keyType, in v: valType): bool {
+    proc add(in k: keyType, in v: valType): bool lifetime this < v {
```
Comment (dlongnecke-cray) [inline modules/standard/Map.chpl:464]: I think we should also have a check for `this < k` as well.

Comment (daviditen) [reply modules/standard/Map.chpl:464]: I agree that that makes sense, but would like to do it as a separate PR if that's OK with you.

Comment (dlongnecke-cray) [reply modules/standard/Map.chpl:464]: Is the idea there that you'll add checks for borrowed keys at that time as well?

Comment (daviditen) [reply modules/standard/Map.chpl:464]: Yes, I think adding tests would need to be a part of that if not earlier.

**Comment by daviditen** [comment]
Thanks!

**Pull request merged**